analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/DoNotHardcodeCredentials_DefaultValues.cs

```diff
@@ -121,18 +121,20 @@ public void Concatenations(string arg)
             a = "Server = localhost;" + " Database = Test; User = SA; " + passwordPrefixVariable + secretConst;     // Noncompliant
             a = "Server = localhost;" + " Database = Test; User = SA; Password = " + secretConst + " suffix";       // Noncompliant
             //  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-            a = someMethod() + " Database = Test; User = SA; Password = " + secretConst + " suffix";                // Noncompliant
+            a = SomeMethod() + " Database = Test; User = SA; Password = " + secretConst + " suffix";                // Noncompliant
             a = "Server = localhost; Database = Test; User = SA; Password = " + secretConst + arg + " suffix";      // Noncompliant
             a = "Server = localhost; Database = Test; User = SA; Password = " + arg + secretConst + " suffix";      // Compliant
             a = secretConst + "Server = localhost; Database = Test; User = SA; Password = " + arg;                  // Compliant
-            a = "Server = localhost; Database = Test; User = SA; " + someMethod() + secretConst;                    // Compliant
+            a = "Server = localhost; Database = Test; User = SA; " + SomeMethod() + secretConst;                    // Compliant
 
             // Reassigned
+            arg += "Literal";
```

Comment:
could you please also add a test with `??=`?
```
var a = SomeMethod();
a ??= "hardcoded";
...
```

I'm not sure how this should behave. It's the opposite of initializeToConstant then assignValueFromMethod

could you check with AppSec before merging this, please? it's a Security Hotspot so it should show possibly vulnerable cases, and this looks like one.

Reply:
I've added a test case. The `??=` is semantically same as this:
```
string SomeMethod() => null;

var a = SomeMethod();
a += "hardcoded";
```
I don't think we need to check with appsec. This is not SE rule and such constant propagation detection wasn't available until recently. So killing the noise for this and other rules should be priority.

Reply:
I don't think it has the same semantics. The `+=` "enhances" a non-hardcoded password (like adding a salt to it), whereas "??=" replaces sets a hardcoded value to it.

But I agree we it's the same as
```
var password = "hardcoded";
var b = someMethod();
if (b != null) 
{
  password = b;
}
```
so I drop my case.

Reply:
I drop my case especially as indeed it's out of scope to change the implementation.

I think you can send an FYI to AppSec just for them to know of this behavior (after merging this PR)